src/Miso/Style.hs

```diff
@@ -0,0 +1,1524 @@
+  , relativeLeftOf
+  , relativeRightOf
+  , relativeTopOf
+  , right
+  , rowGap
+  , textAlign
+  , textDecoration
+  , textIndent
+  , textOverflow
+  , textShadow
+  , textStrokeColor
+  , textStroke
+  , textStrokeWidth
+  , top
+  , transform
+  , transformOrigin
+  , transitionDelay
+  , transitionDuration
+  , transition
+  , transitionProperty
+  , transitionTimingFunction
+  , verticalAlign
+  , visibility
+  , whiteSpace
+  , width
+  , wordBreak
+  , xAutoFontSize
+  , xAutoFontSizePresetSizes
+  , xHandleColor
+  , xHandleSize
+  , zIndex
+  -- *** Colors
+  , module Miso.Style.Color
+  ) where
+-----------------------------------------------------------------------------
+import           Data.Map (Map)
+import qualified Data.Map as M
+import           Miso.String (MisoString)
+import qualified Miso.String as MS
+import           Miso.Style.Color
+import           Miso.Property
+import           Miso.Types (Attribute)
+import qualified Miso.Types as MT
+-----------------------------------------------------------------------------
+import           Prelude hiding (filter)
+-----------------------------------------------------------------------------
+-- | Type for a CSS 'Style'
+--
+(===) :: MisoString -> MisoString -> Style
+(===) = (,)
+-----------------------------------------------------------------------------
+-- | 'Style'
+--
+-- Type for a CSS 'Style'
+--
+type Style = (MisoString, MisoString)
+-----------------------------------------------------------------------------
+-- | 'Styles'
+--
+-- Type for a @Map@ of CSS 'Style'. Used with @StyleSheet@.
+--
+type Styles = Map MisoString MisoString
+-----------------------------------------------------------------------------
+-- | 'StyleSheet'
+--
+-- Type for a CSS style on native
+--
+newtype StyleSheet = StyleSheet { getStyleSheet :: Map MisoString Styles }
+-----------------------------------------------------------------------------
+sheet_ :: [(MisoString, Styles)] -> StyleSheet
+sheet_ = StyleSheet . M.fromList
+-----------------------------------------------------------------------------
+-- | @style_@ is an attribute that will set the @style@
+-- attribute of the associated DOM node to @attrs@.
+--
+-- @style@ attributes not contained in @attrs@ will be deleted.
+--
+-- > import qualified Data.Map as M
+-- > div_ [ style_ [ backgroundColor "red" ] [ ]
+--
+-- <https://developer.mozilla.org/en-US/docs/Web/CSS>
+--
+style_ :: [Style] -> Attribute action
+style_ = MT.Styles . M.fromList
+-----------------------------------------------------------------------------
+-- | Set "style" property
+--
+-- > view m = div_ [ styleInline_ "background-color:red;color:blue;" ] [ "foo" ]
+--
+-- https://developer.mozilla.org/en-US/docs/Web/CSS
+styleInline_ ::  MisoString -> Attribute action
+styleInline_ = textProp "style"
+-----------------------------------------------------------------------------
+-- | 'renderStyleSheet'
+--
+-- Renders a 'StyleSheet' to a 'MisoString'
```

Comment:
Suggestion: I think adding link with the function name in the haddocks of that function only adds noise to the haddock (nobody is going to every click on the link to function X to navigate to function X if that function is just below it). I would suggest removing these. If you search for "-- | '" it's 155x in this PR :sweat_smile: 
```suggestion
-- | Renders a 'StyleSheet' to a 'MisoString'
```